p2pp/mcf.py

```diff
@@ -634,7 +647,8 @@ def gcode_parseline(index):
 
     if g.is_movement_command():
 
-        if v.retract_move and g.E < 0:
+        if v.retract_move and g.is_retract_command():
+            # This is going to break stuff, G10 cannot take X and Y, what to do?
```

Comment:
We'll never hit this line of code with firmware retractions enabled. Since G10 and G11 are not movement commands, but it will break RRF gcode if we try to set X and Y values on a G10 command. I am not yet sure what to do in this case.

Reply:
Perhaps detecting before the processing move commands and inserting a G0 to the retract_x and retract_y locations.